static/src/javascripts/projects/common/modules/discussion/comments.js

```diff
@@ -198,26 +200,29 @@ Comments.prototype.fetchComments = function(options) {
 
 
 Comments.prototype.renderComments = function(resp) {
-
     // The resp object received has a collection of rendered html fragments, ready for DOM insertion.
     // - commentsHtml - the main comments content.
     // - paginationHtml - the discussion's pagination based on user page size and number of comments.
     // - postedCommentHtml - an empty comment for when the user successfully posts a comment.
+    return new Promise(function (resolve) {
```

Comment:
or you could use `fastdom-promise` so that your DOM manipulations are inside a fastdom.write